modular_skyrat/modules/credits/end.dm

```diff
@@ -0,0 +1,192 @@
+#define CREDIT_ROLL_SPEED 185
+#define CREDIT_SPAWN_SPEED 20
+#define CREDIT_ANIMATE_HEIGHT (14 * world.icon_size)
+#define CREDIT_EASE_DURATION 22
+GLOBAL_VAR_INIT(end_credits_song, null)
+GLOBAL_VAR_INIT(end_credits_title, null)
+GLOBAL_LIST(end_titles)
+#define JOINTEXT(X) jointext(X, null)
+/datum/controller/subsystem/ticker/declare_completion()
+	for(var/client/C)
+		if(!C.credits)
+			C.RollCredits()
+	for(var/thing in GLOB.clients)
+		var/client/C = thing
+		if (!C)
+			continue
+	. = ..()
+
+/datum/controller/subsystem/ticker/show_roundend_report(client/C, report_type = null)
+	. = ..()
+	for(var/iteration in GLOB.clients)
+		var/client/clients = iteration
+		if(!clients.credits)
+			clients.RollCredits()
+/* /client
+	var/list/credits */
+
+
+/client/proc/RollCredits()
+	set waitfor = FALSE
+
+
+	if(!GLOB.end_titles)
+		GLOB.end_titles = generate_titles()
+
+	LAZYINITLIST(credits)
+
+/* 	if(mob)
+		mob.overlay_fullscreen("meeting",/atom/movable/screen/fullscreen/emergency_meeting) */
+//	sound_to_playing_players('modular_skyrat/master_files/sound/music/elibao.ogg', volume = 100, vary = FALSE, frequency = 0, channel = 0, pressure_affected = FALSE)
+
+	var/list/_credits = credits
+	verbs += /client/proc/ClearCredits
+	for(var/I in GLOB.end_titles)
+		if(!credits)
+			return
+		var/atom/movable/screen/credit/T = new(null, I, src)
+		_credits += T
+		T.rollem()
+		sleep(CREDIT_SPAWN_SPEED)
+	sleep(CREDIT_ROLL_SPEED - CREDIT_SPAWN_SPEED)
+
+	ClearCredits()
+	verbs -= /client/proc/ClearCredits
+
+/client/proc/ClearCredits()
+	set name = "Stop End Titles"
+	set category = "OOC"
+	verbs -= /client/proc/ClearCredits
+	QDEL_NULL(credits)
+//	mob.clear_fullscreen("meeting")
+
+/atom/movable/screen/credit
+	icon_state = "blank"
+	mouse_opacity = 0
+	alpha = 0
+	screen_loc = "CENTER-7,CENTER-7"
+	plane = HUD_PLANE
+	layer = ABOVE_ALL_MOB_LAYER
+	var/client/parent
+	var/matrix/target
+
+/atom/movable/screen/credit/Initialize(mapload, credited, client/P)
+	. = ..()
+	parent = P
+	maptext = {"<div style="font:'Small Fonts'">[credited]</div>"}
+	maptext_height = world.icon_size * 2
+	maptext_width = world.icon_size * 14
+
+/atom/movable/screen/credit/proc/rollem()
+	var/matrix/M = matrix(transform)
+	M.Translate(0, CREDIT_ANIMATE_HEIGHT)
+	animate(src, transform = M, time = CREDIT_ROLL_SPEED)
+	target = M
+	animate(src, alpha = 255, time = CREDIT_EASE_DURATION, flags = ANIMATION_PARALLEL)
+	spawn(CREDIT_ROLL_SPEED - CREDIT_EASE_DURATION)
+		if(!QDELETED(src))
+			animate(src, alpha = 0, transform = target, time = CREDIT_EASE_DURATION)
+			sleep(CREDIT_EASE_DURATION)
+			qdel(src)
+	parent.screen += src
+
+/atom/movable/screen/credit/Destroy()
+	var/client/P = parent
```

Comment:
`parent` is already a typecasted var on `/screen/credit`, also no one-letter vars